util/MeasurePopulations.ts

```diff
@@ -33,16 +29,41 @@ export function getMeasurePopulationsForSelection(measure: fhir4.Measure): Multi
       !measurePopulations.map(p => p.value).includes(populationCode) &&
       !EXCLUDED_MEASURE_POPULATIONS.find(p => p === populationCode)
     ) {
+      const labelAdjust = labelAdjustment(population, measure.group?.[0]);
+      const label = (populationCriteriaExpression || populationDisplay || populationCode) + labelAdjust;
       measurePopulations.push({
-        value: populationCode,
-        label: populationCriteriaExpression || populationDisplay || populationCode,
+        value: populationCode + labelAdjust,
+        label: label,
         disabled: false
       });
     }
   });
   return measurePopulations;
 }
 
+/**
+ * Creates a label string that can be attached to the default (population criteria, etc.) and adjusts to show measure
+ * observation related population reference so that the population string is unique across different measure observations
+ * @param {fhir4.MeasureGroupPopulation} population - FHIR population from a measure group
+ * @param {fhir4.MeasureGroup} group - FHIR measure group for observation population lookup
+ * @returns {string} adjustment string
+ */
+export function labelAdjustment(
+  population: fhir4.MeasureGroupPopulation,
+  group: fhir4.MeasureGroup | undefined
+): string {
+  let label = ''; // default adjustment is none
+  const criteriaReference = population?.extension?.find(e => {
+    return e?.url === 'http://hl7.org/fhir/us/cqfmeasures/StructureDefinition/cqfm-criteriaReference';
+  })?.valueString;
+  // handle measure observation
+  if (group && population.code?.coding?.[0].code === 'measure-observation' && criteriaReference) {
+    const obsPop = group.population?.find(p => p && p.id === criteriaReference);
```

Comment:
```suggestion
    const obsPop = group.population?.find(p => p.id === criteriaReference);
```
I tested it and it seems to work?